### PR TITLE
Revise data docs and remove links to vega wiki. Fixes #2851

### LINKED ISSUE #2851: Data Docs
- [x] Missing information about how Vega ingests data. (We should copy from `https://vega.github.io/vega/docs/data/`)
- [x] Final Pass

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: site/docs/data.md [hunk 1]
```diff
@@ -45,7 +43,7 @@ Here is a list of all properties describing a `data` source from URL:
 
 {% include table.html props="url,format" source="UrlData" %}
 
-For example, the following specification loads data from a relative `url`: `data/cars.json`. Note that the format type is implicitly json by default.
+For example, the following specification loads data from a relative `url`: `data/cars.json`. Note that the format type is implicitly `json` by default.
```
Comment (kanitw) [inline site/docs/data.md]: > Note that the format type is implicitly `json` by default.

Didn't we read extension name or something? 

In any case we use `"json"` for string values and `key` for keys.

File: site/docs/data.md [hunk 2]
```diff
@@ -5,20 +5,12 @@ title: Data
...
-  "mark": ... ,
-  "encoding": ... ,
-  ...
-}
-```
+Akin to [Vega](https://www.github.com/vega/vega)'s [data model](https://vega.github.io/vega/docs/data/), the basic data model used by Vega-Lite is *tabular* data, similar to a spreadsheet or a database table. Individual data sets are assumed to contain a collection of records, which may contain any number of named data fields.
 
 Vega-Lite's `data` property describes the visualization's data source as part of the specification, which can be either [inline data](#inline) (`values`) or [a URL from which to load the data](#url) (`url`).  Alternatively, we can create an empty, [named data source](#named) (`name`), which can be [bound at runtime](https://vega.github.io/vega/docs/api/view/#data).
 
+Upon ingest, Vega assigns each data object a unique id property, accessible via a custom [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol). As a result, the id property is not accessible via a string key and is not enumerable, though you can observe the id value when inspecting data objects in a JavaScript console.
```
Comment (kanitw) [inline site/docs/data.md]: Vega seems to come out of nowhere?  Need 1-2 sentence to provide more context?

Comment (domoritz) [inline site/docs/data.md]: Vega is introduced in the paragraph above.

File: site/docs/data.md (hunk 1 above)
Comment (domoritz) [inline site/docs/data.md]: Fixed

File: site/docs/data.md (hunk 2 above)
Comment (kanitw) [inline site/docs/data.md]: Ah true, but it might be unclear here that Vega is the underlying engine.  (We only say "akin to Vega" above.)

Comment (domoritz) [inline site/docs/data.md]: I think that needs to be clear from the beginning. Saying that everywhere makes the whole docs unreadable. I think we should leave it.

Comment (kanitw) [inline site/docs/data.md]: >that needs to be clear from the beginning

I agree.  Maybe change the first paragraph?

Comment (domoritz) [inline site/docs/data.md]: This is something about all the docs so it should be in one place. I want to keep the docs lean so that people actualy read it.

Comment (kanitw) [inline site/docs/data.md]: Reading one more time, It seems like this paragraph is quite irrelevant I'll just remove it.

Comment (domoritz) [inline site/docs/data.md]: Agreed.

**Pull request merged**